src/components/SplitHorizontal/SplitHorizontal.jsx

```diff
@@ -0,0 +1,435 @@
+		const {
+			isExpanded,
+			collapseShift,
+			onResize,
+		} = this.props;
+
+		const {
+			secondaryRef,
+			secondary,
+			bottom,
+			primaryRef,
+		} = this.panes;
+
+		onResize(
+			this.applyDeltaToSecondaryHeight(dY, isExpanded, this.secondaryStartRect, secondaryRef, secondary, bottom, this.storedRefs.inner, primaryRef, collapseShift),
+			{ props: this.props, event }
+		);
+
+		this.resetAnimation(this.storedRefs.inner, secondaryRef, primaryRef);
+	},
+
+	componentWillReceiveProps(nextProps) {
+		const {
+			isAnimated,
+			isExpanded,
+			collapseShift,
+		} = nextProps;
+
+		const {
+			secondaryRef,
+		} = this.getPanes();
+
+		if (this.props.isExpanded && !isExpanded) { // collapse secondary
+			const secondaryRect = secondaryRef.getBoundingClientRect();
+			this.collapseSecondary(secondaryRect.height - collapseShift);
+		} else if (!this.props.isExpanded && isExpanded) { // expand secondary
+			this.expandSecondary();
+		}
+
+		if (this.state.isAnimated !== isAnimated) {
+			this.setState({
+				isAnimated,
+			});
+		}
+	},
+
+	componentDidMount() {
+		const {
+			isExpanded,
+			isAnimated,
+			collapseShift,
+		} = this.props;
+
+		const {
+			primaryRef,
+			secondaryRef,
+		} = this.getPanes();
+
+		const {
+			inner,
+		} = this.storedRefs;
+
+		_.defer(() => {
+			this.disableAnimation(inner, secondaryRef, primaryRef);
+			inner.style.visibility = 'hidden';
+			_.defer(() => {
+				if (!isExpanded) { // collapse secondary
+					const secondaryRect = secondaryRef.getBoundingClientRect();
+					this.collapseSecondary(secondaryRect.height - collapseShift);
+				}
+
+				_.defer(() => {
+					if (isAnimated) {
+						this.setState({ isAnimated });
+					}
+					_.defer(() => {
+						inner.style.visibility = '';
+						this.resetAnimation(inner, secondaryRef, primaryRef);
+					});
+				});
+			});
+		});
+	},
+
+	storeRef(name) {
+		return (ref) => {
+			this.storedRefs[name] = ref;
+		};
+	},
+
+	componentWillMount() {
+		this.storedRefs = {};
+	},
+
+	render() {
+		const {
+			children,
+			className,
+			...passThroughs,
+		} = this.props;
+
+		const {
+			isAnimated,
+			isExpanded,
+			collapseAmount,
+		} = this.state;
+
+		const {
+			top: topPaneProps,
+			bottom: bottomPaneProps,
+			secondary,
+		} = this.getPanes();
+
+		const dividerProps = _.get(_.first(filterTypes(children, SplitHorizontal.Divider)), 'props', {});
+
+		let from, to;
+
+		if (!isExpanded) {
+			from = { slideAmount: 0 };
+			to = { slideAmount: collapseAmount };
+		} else {
+			from = { slideAmount: 0 };
+			to = { slideAmount: 0 };
+		}
+
+		const isBottomSecondary = bottomPaneProps === secondary;
+
+		return (
+			<div
+				{...omitProps(passThroughs, SplitHorizontal)}
+				className={cx('&', {
+					'&-is-expanded': isExpanded,
+					'&-is-animated': isAnimated,
+				}, className)}
+			>
+				<Motion defaultStyle={from} style={isAnimated ? _.mapValues(to, (val) => (spring(val, {stiffness: 640, damping: 37, precision: 1}))) : to}>
```

Comment:
Are these values re-used in multiple places? Should probably have constants if that's the case.